fastapi/models/fastapi_endpoint_demo.py

```diff
@@ -0,0 +1,190 @@
+# Copyright 2022 ACSONE SA/NV
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/LGPL).
+from enum import Enum
+from typing import List
+
+from odoo import _, api, fields, models
+from odoo.api import Environment
+from odoo.exceptions import AccessError, MissingError, UserError, ValidationError
+
+from odoo.addons.base.models.res_partner import Partner
+
+from fastapi import APIRouter, Depends, HTTPException, status
+from fastapi.security import APIKeyHeader
+from pydantic import BaseModel, Field
+
+from ..depends import (
+    authenticated_partner,
+    authenticated_partner_from_basic_auth_user,
+    authenticated_partner_impl,
+    fastapi_endpoint,
+    odoo_env,
+)
+
+
+class FastapiEndpoint(models.Model):
+
+    _inherit = "fastapi.endpoint"
+
+    app: str = fields.Selection(
+        selection_add=[("demo", "Demo Endpoint")], ondelete={"demo": "cascade"}
+    )
+    demo_auth_method = fields.Selection(
+        selection=[("api_key", "Api Key"), ("http_basic", "HTTP Bacic")],
+        string="Authenciation method",
+    )
+
+    def _get_fastapi_routers(self) -> List[APIRouter]:
```

Comment:
```suggestion
    def get_fastapi_routers(self) -> List[APIRouter]:
```
IMHO, the (documentation)[https://github.com/acsone/rest-framework/blob/16.0-fastapi/fastapi/README.rst?plain=1#L222] is correct and this code must be changed